common/libs/transaction.ts

```diff
@@ -71,13 +71,63 @@ export function getTransactionFields(tx: EthTx) {
   };
 }
 
-export async function generateCompleteTransactionFromRawTransaction(
+function getValue(token: Token | null | undefined, tx: ExtendedRawTransaction) {
+  let value;
+  if (token) {
+    value = new Big(ERC20.$transfer(tx.data).value);
+  } else {
+    value = new Big(tx.value);
+  }
+  return value;
+}
+
+async function getBalance(
   node: INode,
   tx: ExtendedRawTransaction,
-  wallet: IWallet,
   token: Token | null | undefined
-): Promise<CompleteTransaction> {
-  const { to, data, gasLimit, gasPrice, from, chainId, nonce } = tx;
+) {
+  const { from } = tx;
+  const ETHBalance = await node.getBalance(from);
+  let balance;
```

Comment:
Can make balance a constant, with its value being the return value of a ternary. Would make it more legible.